compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2332,17 +2332,19 @@ public void visit(BLangXMLElementLiteral bLangXMLElementLiteral) {
         SymbolEnv xmlElementEnv = SymbolEnv.getXMLElementEnv(bLangXMLElementLiteral, env);
 
         // Visit in-line namespace declarations
-        bLangXMLElementLiteral.attributes.forEach(attribute -> {
-            if (attribute.name.getKind() == NodeKind.XML_QNAME
-                    && ((BLangXMLQName) attribute.name).prefix.value.equals(XMLConstants.XMLNS_ATTRIBUTE)) {
+        for (BLangXMLAttribute attribute : bLangXMLElementLiteral.attributes) {
+            if (attribute.name.getKind() == NodeKind.XML_QNAME && isXmlNamespaceAttribute(attribute)) {
+                BLangXMLQuotedString value = attribute.value;
+                if (value.getKind() == NodeKind.XML_QUOTED_STRING && value.textFragments.size() > 1) {
+                    dlog.error(value.pos, DiagnosticCode.INVALID_XML_NS_INTERPOLATION);
+                }
                 checkExpr(attribute, xmlElementEnv, symTable.noType);
             }
-        });
+        }
 
         // Visit attributes.
         bLangXMLElementLiteral.attributes.forEach(attribute -> {
-            if (attribute.name.getKind() != NodeKind.XML_QNAME
-                    || !((BLangXMLQName) attribute.name).prefix.value.equals(XMLConstants.XMLNS_ATTRIBUTE)) {
+            if (attribute.name.getKind() != NodeKind.XML_QNAME || !isXmlNamespaceAttribute(attribute)) {
                 checkExpr(attribute, xmlElementEnv, symTable.noType);
             }
```

Comment:
is it not possible to add this to the 'else' block of the if block at line no. 2336 and remove this?

Reply:
It turned out that first iteration defines the xml namespaces, and the second iteration depends on those previously defined namespaces, there for it's not possible to merge the two into single foreach statement.